A couple of doc tweaks (#603)
dbos-inc/dbos-docs: docs/faq.md

The [architecture page](./architecture.md) describes how to architect a distributed DBOS application and how DBOS scales.

### When should I use the DBOS Conductor?

> Conductor is required for applications that use more than one process. In addition to cross-process recovery, Conductor offers data retention, observability, cluster-wide metrics, alerts, and organization management with RBAC. These features are designed to help engineering teams maintain long-term app health and reliability. We recommend using Conductor whenever you are running in a production environment. Conductor is not needed for dev, tests, prototypes, or hobbyist use cases.

Reviewer: ```suggestion
Conductor is strongly recommended for applications that use more than one process. In addition to cross-process recovery, Conductor offers data retention, observability, cluster-wide metrics, alerts, and organization management with RBAC. These features are designed to help engineering teams maintain long-term app health and reliability. We recommend using Conductor whenever you are running in a production environment. 
```
